### PR TITLE
[PSDK] extend struct _SECPKG_DLL_FUNCTIONS, fix definitions

### PR DESCRIPTION
**Pull request by andy-123**
- add missing enums to MSV1_0_PROTOCOL_MESSAGE_TYPE and SECPKG_EXTENDED_INFORMATION_CLASS
- add missing function entries to LSA_SECPKG_FUNCTION_TABLE and add needed definitions.
- fix params of PLSA_AP_CALL_PACKAGE
- fix SpInitializeFn (+ NTAPI)

### CODE REVIEW

File: sdk/include/psdk/ntsecapi.h [hunk 1]
```diff
@@ -228,7 +229,8 @@ typedef enum _MSV1_0_PROTOCOL_MESSAGE_TYPE {
   MsV1_0DeriveCredential,
   MsV1_0CacheLookup,
   MsV1_0SetProcessOption
-} MSV1_0_PROTOCOL_MESSAGE_TYPE, *PMSV1_0_PROTOCOL_MESSAGE_TYPE;
+} MSV1_0_PROTOCOL_MESSAGE_TYPE,
+    *PMSV1_0_PROTOCOL_MESSAGE_TYPE;
```
Comment (HBelusca) [inline sdk/include/psdk/ntsecapi.h]: broken.

File: sdk/include/psdk/ntsecpkg.h [hunk 2]
```diff
@@ -276,11 +279,11 @@ typedef NTSTATUS (NTAPI *PLSA_AUDIT_ACCOUNT_LOGON)(ULONG, BOOLEAN,
...
-typedef void (NTAPI *PLSA_PROTECT_MEMORY)(PVOID, ULONG);
-typedef NTSTATUS (NTAPI *PLSA_OPEN_TOKEN_BY_LOGON_ID)(PLUID, HANDLE *);
-typedef NTSTATUS (NTAPI *PLSA_EXPAND_AUTH_DATA_FOR_DOMAIN)(PUCHAR, ULONG, PVOID, PUCHAR *, PULONG);
-typedef PVOID (NTAPI *PLSA_ALLOCATE_PRIVATE_HEAP)(SIZE_T);
-typedef void (NTAPI *PLSA_FREE_PRIVATE_HEAP)(PVOID);
+typedef void(NTAPI *PLSA_PROTECT_MEMORY)(PVOID, ULONG);
+typedef NTSTATUS(NTAPI *PLSA_OPEN_TOKEN_BY_LOGON_ID)(PLUID, HANDLE *);
+typedef NTSTATUS(NTAPI *PLSA_EXPAND_AUTH_DATA_FOR_DOMAIN)(PUCHAR, ULONG, PVOID, PUCHAR *, PULONG);
+typedef PVOID(NTAPI *PLSA_ALLOCATE_PRIVATE_HEAP)(SIZE_T);
+typedef void(NTAPI *PLSA_FREE_PRIVATE_HEAP)(PVOID);
```
Comment (HBelusca) [inline sdk/include/psdk/ntsecpkg.h]: Broken.

File: sdk/include/psdk/ntsecapi.h (hunk 1 above)
Comment (andy-123) [reply sdk/include/psdk/ntsecapi.h]: Travis CI wants it that way ... should i revert and ignore travis?

Comment (HBelusca) [reply sdk/include/psdk/ntsecapi.h]: Yes at this moment we can ignore it.

Comment (andy-123) [reply sdk/include/psdk/ntsecapi.h]: I've reverted the code style fixes. (added as new commit).

File: sdk/include/psdk/ntsecpkg.h [hunk 3]
```diff
@@ -256,6 +276,11 @@ typedef NTSTATUS (NTAPI *PLSA_AUDIT_ACCOUNT_LOGON)(ULONG, BOOLEAN,
  PUNICODE_STRING, PUNICODE_STRING, PUNICODE_STRING, NTSTATUS);
 typedef NTSTATUS (NTAPI *PLSA_CALL_PACKAGE_PASSTHROUGH)(PUNICODE_STRING, PVOID,
  PVOID, ULONG, PVOID*, PULONG, PNTSTATUS);
+typedef void (NTAPI *PLSA_PROTECT_MEMORY)(PVOID, ULONG);
+typedef NTSTATUS (NTAPI *PLSA_OPEN_TOKEN_BY_LOGON_ID)(PLUID, HANDLE *);
+typedef NTSTATUS (NTAPI *PLSA_EXPAND_AUTH_DATA_FOR_DOMAIN)(PUCHAR, ULONG, PVOID, PUCHAR *, PULONG);
+typedef PVOID (NTAPI *PLSA_ALLOCATE_PRIVATE_HEAP)(SIZE_T);
+typedef void (NTAPI *PLSA_FREE_PRIVATE_HEAP)(PVOID);
```
[the remark this answers is not in the document]
Comment (andy-123) [reply sdk/include/psdk/ntsecpkg.h]: fixed

File: sdk/include/psdk/ntsecpkg.h (hunk 2 above)
Comment (andy-123) [reply sdk/include/psdk/ntsecpkg.h]: dito.

File: sdk/include/psdk/ntsecpkg.h [hunk 4]
```diff
@@ -256,6 +276,11 @@ typedef NTSTATUS (NTAPI *PLSA_AUDIT_ACCOUNT_LOGON)(ULONG, BOOLEAN,
  PUNICODE_STRING, PUNICODE_STRING, PUNICODE_STRING, NTSTATUS);
 typedef NTSTATUS (NTAPI *PLSA_CALL_PACKAGE_PASSTHROUGH)(PUNICODE_STRING, PVOID,
  PVOID, ULONG, PVOID*, PULONG, PNTSTATUS);
+typedef void (NTAPI *PLSA_PROTECT_MEMORY)(PVOID, ULONG);
```
[the remark this answers is not in the document]
Comment (andy-123) [reply sdk/include/psdk/ntsecpkg.h]: fixed.

File: sdk/include/psdk/ntsecapi.h [hunk 5]
```diff
@@ -226,7 +226,8 @@ typedef enum _MSV1_0_PROTOCOL_MESSAGE_TYPE {
   MsV1_0CacheLogon,
   MsV1_0SubAuth,
   MsV1_0DeriveCredential,
-  MsV1_0CacheLookup
+  MsV1_0CacheLookup,
+  MsV1_0SetProcessOption
```
Comment (ThFabba) [inline sdk/include/psdk/ntsecapi.h]: Should guard this with `_WIN32_WINNT >= _WIN32_WINNT_WINXP` according to latest MS headers.

File: sdk/include/psdk/ntsecpkg.h [hunk 6]
```diff
@@ -152,6 +154,22 @@ typedef struct _SECPKG_MUTUAL_AUTH_LEVEL {
     ULONG MutualAuthLevel;
 } SECPKG_MUTUAL_AUTH_LEVEL, *PSECPKG_MUTUAL_AUTH_LEVEL;
 
+typedef struct _SECPKG_WOW_CLIENT_DLL {
+    UNICODE_STRING WowClientDllPath;
```
Comment (ThFabba) [inline sdk/include/psdk/ntsecpkg.h]: `SECURITY_STRING`?

Comment (HBelusca) [reply sdk/include/psdk/ntsecpkg.h]: Do we use that type elsewhere in this file?

File: sdk/include/psdk/ntsecapi.h (hunk 5 above)
Comment (andy-123) [reply sdk/include/psdk/ntsecapi.h]: fixed

File: sdk/include/psdk/ntsecpkg.h (hunk 6 above)
Comment (andy-123) [reply sdk/include/psdk/ntsecpkg.h]: No. But its defined as SECURITY_STRING in MS Headers and should be used in some other places. AFAICS SECURITY_STRING is the same as UNICODE_STRING in 32 bit mode. It differs in 64 bit.
I've added a definition to psdk/sspi.h for SECURITY_STRING.
Further i've added a commit that corrects wrong definitions in ntsecpkg.h regarding SECURITY_STRING.

**Pull request merged**